instrumentation/opentelemetry-instrumentation-sklearn/src/opentelemetry/instrumentation/sklearn/__init__.py

```diff
@@ -0,0 +1,738 @@
+              estimator.
+            method_name (str): The method name of the estimator on which to
+              apply a span.
+       """
+        orig_method_name = "_otel_original_" + method_name
+        if isclass(estimator):
+            qualname = estimator.__qualname__
+        else:
+            qualname = estimator.__class__.__qualname__
+        delegator = get_delegator(estimator, method_name)
+        if self._check_instrumented(estimator, method_name):
+            logger.debug(
+                "Uninstrumenting: %s.%s", qualname, method_name,
+            )
+            _, orig_method = getattr(estimator, orig_method_name)
+            setattr(
+                estimator, method_name, orig_method,
+            )
+            delattr(estimator, orig_method_name)
+        elif delegator is not None:
+            if not hasattr(delegator, "_otel_original_fn"):
+                logger.debug(
+                    "Already uninstrumented: %s.%s", qualname, method_name,
+                )
+                return
+            setattr(
+                delegator, "fn", getattr(delegator, "_otel_original_fn"),
+            )
+            delattr(delegator, "_otel_original_fn")
+        else:
+            logger.debug(
+                "Already uninstrumented: %s.%s", qualname, method_name,
+            )
+
+    def _uninstrument_instance_method(
+        self, estimator: BaseEstimator, method_name: str
+    ):
+        """Uninstrument an instance method.
+
+        Replaces the patched method with the original, and deletes the
+        attribute which stored the original method.
+
+        Args:
+            estimator (BaseEstimator): A class or instance of an ``sklearn``
+              estimator.
+            method_name (str): The method name of the estimator on which to
+              apply a span.
+       """
+        orig_method_name = "_otel_original_" + method_name
+        if isclass(estimator):
+            qualname = estimator.__qualname__
+        else:
+            qualname = estimator.__class__.__qualname__
+        if self._check_instrumented(estimator, method_name):
+            logger.debug(
+                "Uninstrumenting: %s.%s", qualname, method_name,
+            )
+            _, orig_method = getattr(estimator, orig_method_name)
+            setattr(
+                estimator, method_name, orig_method,
+            )
+            delattr(estimator, orig_method_name)
+        else:
+            logger.debug(
+                "Already uninstrumented: %s.%s", qualname, method_name,
+            )
+
+    def _instrument_class_method(
+        self, estimator: Type[BaseEstimator], method_name: str
+    ):
+        """Instrument an estimator method with a span.
+
+        When instrumenting we attach a tuple of (Class, method) to the
+        attribute ``_original_<method_name>`` for each method. This allows
+        us to replace the patched with the original in unstrumentation, but
```

Comment:
```suggestion
        us to replace the patched with the original in instrumentation, but
```

Reply:
This is actually correct as written. We want to make sure that we _uninstrument_ properly in child classes by replacing patches with originals, only if the method is uninherited.

Reply:
Right, I don't believe `unstrumentation` is an actual word?

Reply:
Ah so you mean `uninstrumentation` as the suggestion, sure.